### ISSUE TITLE
warn when snapshots are 1GB or larger

### ISSUE THREAD

**Issue by srdjanrosic** [sabeechen/hassio-google-drive-backup#420]
feature request

Turns out, snapshots over 1GB are not easily restorable through the home assistant UI when starting from a clean install (e.g. after a dead sd card) - I'm guessing the restore code requires reading an actual file for some reason, sadly.  [edit: they are restorable with slightly more difficulty - you need to setup home assistant just enough to have e.g. samba or sftp or this addon then do a manual upload and can then do a full restore)

With various addons (like ESPHome for example) keeping a bunch of temporary files under /config for each one of the devices, and influxdb, it's easy to end up above the limit. Cleaning up ESPHome helps. ( it gets my own raspberry pi install to 975M :) ; phew - I should probably script ESPHome cleanups)

It'd be nice and probably easy to add some warning icon (pick, not sure which one exactly) in the snapshot list in the UI for these large snapshots.

A better (but perhaps more work) feature, would be to fire a notification into Home Assistant whenever a large snapshot occurs.

The reason, I'm guessing it'd be more work is because I can imagine some folks would probably prefer to run with large snapshots, and not see the notification spam - so maybe it'd need one more config setting.

let me know if this sounds like a reasonable request?

**Comment by sabeechen** [comment]
I'm hesitant to implement this for a handful of reasons:
- Most people using this addon have huge snapshots and wouldn't do anything with the information if I warned them.
- I try to keep the web-ui clean and understandable, only explaining whats critical to know.
- The messaging the onboarding gives about restoring snapshots is pretty good if you try to upload something over 1GB, it links to a page with suggestions about using samba, nfs, etc to upload.
- I think they're going to fix this in the near-ish future.  The onboarding process is something the HA folks are still working out the kinks on.

I did a little digging to track down why the onboarding works this way.  The relevant code is [here](https://github.com/home-assistant/core/blob/dev/homeassistant/components/hassio/http.py#L84).  It looks like originally there was a 10s time limit for uploading snapshots, which obviously wouldn't work well for anything over a few 10's of Mb.  Then [this commit](https://github.com/home-assistant/core/pull/43851) upped the limit to 5 minutes to try and address that.  So there are still some big problems with it:
- The HA frontend appears to proxy requests to the supervisor by reading the entire request into memory.  I suspect thats the reason for the 1 GB cap, which also isn't a good idea.
- 5 minutes still won't be a long enough timeout if you have a slow disk (eg an sd card on an rpi) or slow network (eg the shitty wifi antennae on an rpi).

In principle I don't see why the request couldn't just be streamed to the supervisor.  I might work on that if I can figure out how to set up a dev environment for HA.  Regardless if I or someone else doesn't fix this in the near future I'll think about how to add better messaging for it in the addon.  For now I'm going to leave it be.
